### DISCUSSION TITLE
Pre-bundled configuration file for e.g. TypeScript configuration.

### DISCUSSION THREAD

**Discussion by 7086cmd** [rolldown/rolldown#1973] [Ideas]
Envision a scenario where we need to pre-bundle certain static fields to enhance the runtime performance by minimizing static parsing (similar to compiling a static schema). Given that Rolldown is a fast compiler, I propose pre-bundling the configuration file at high speed and subsequently compiling the project. Additionally, we can include an option for pre-bundling.

**Comment by hyfdev** [comment]
We have discuss this before and my thought has not changed. https://github.com/rolldown/rolldown/pull/610#issuecomment-2009485012.

From my point of view, looking into current frontend technology stack , the main responsibility of supporting `rolldown.config.ts` is not in rolldown itself.

If rolldown is running in `ts-node`, than `rolldown -c rolldown.config.ts` got supported automatically.

I'm not saying that rolldown refuse to support `rolldown.config.ts`. But the current solutions all has their trade offs and we don't want to deal with that kind of complexity in the current timing.

**Reply by hyfdev** [reply]
Other than above thoughts, bundling config file and pass it to rolldown could be done in user-land. If users require this behavior to run rolldown, it could be done in user-land.

It's not something that have to be supported in rolldown's internal code. But Still I'm open to support it if we find the right way to do it.

**Reply by 7086cmd** [reply]
The concept arises when considering the possibility of implementing a mechanism to preserve static content that users may desire to “freeze” when bundling. For instance, during the development of the CLI options schema, we could make the schema static within the bundled file to optimize runtime performance.

**Reply by ikkz** [reply]
Another possible workaround for `rolldown.config.ts` is using [Type stripping](https://nodejs.org/api/typescript.html#type-stripping) in Node v22.6.0. Although it has limited TypeScript features, it is enough in configuration file scenario

**Reply by 7086cmd** [reply]
I attempted to implement it using the `vm` method. Regrettably, it exhibits a significant performance degradation when we pre-bundle the configuration and execute the bundled file within the virtual machine. Is there a more effective approach to loading it?

![prebundling](https://github.com/user-attachments/assets/806ec13e-432e-4674-85e2-b8a4383f7b95)
